src/Microsoft.AspNetCore.SystemWebAdapters.Authentication/RemoteAuthenticationService.cs

```diff
@@ -0,0 +1,131 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+/// <summary>
+/// Service for authenticating a user by making an HTTP request on their behalf to a remote app.
+/// </summary>
+public class RemoteAuthenticationService : IRemoteAuthenticationService
```

Comment:
services have for now been kept internal with only the interfaces (as needed) being exposed. Is there a reason to expose this service itself?